--- virt-top-1.0.4.old/virt-top/virt_top.ml	2012-09-28 16:42:18.525858910 +0100
+++ virt-top-1.0.4/virt-top/virt_top.ml	2012-09-28 16:51:29.470466846 +0100
@@ -222,6 +223,27 @@
   (* Read the init file. *)
   let try_to_read_init_file filename =
     let config = read_config_file filename in
+    (* Replacement functions that raise better errors when
+     * parsing the init file.
+     *)
+    let int_of_string s =
+      try int_of_string s
+      with Invalid_argument _ ->
+        eprintf (f_"%s: could not parse '%s' in init file: expecting an integer")
+          filename s;
+        exit 1 in
+    let float_of_string s =
+      try float_of_string s
+      with Invalid_argument _ ->
+        eprintf (f_ "%s: could not parse '%s' in init file: expecting a number")
+          filename s;
+        exit 1 in
+    let bool_of_string s =
+      try bool_of_string s
+      with Invalid_argument _ ->
+        eprintf (f_ "%s: could not parse '%s' in init file: expecting %s")
+          filename s "true|false";
+        exit 1 in
     List.iter (
       function
       | _, "display", mode -> display_mode := display_of_cli mode
